File: readability.c
#include <stdio.h>
#include <cs50.h>
#include <string.h>
#include <ctype.h>
#include <math.h>

// int index(float l, float s, float w);

int main(void){
    string text = get_string("Text: ");
    int tLength = strlen(text);
    int sent = 0;
    int word = 0;
    int let = 0;
    
    if (isalnum(text[0]))
    {
        word = 1;
    }
    
    for (int i = 0; i < tLength; i++)
    {
        if (isalpha(text[i]))
        {
            let++;
        }
        if (i < tLength - 1 && isspace(text[i]) && isalnum(text[i + 1]))
        {
            word++;
        }
        if (i > 0 && (text[i] == '.' || text[i] == '!' || text[i] == '?' ) && isalnum(text[i - 1]))
        {
            sent++;
        }
    }
    
    int grade = 0.0588 * (100 * let / word) - 0.296 * (100 * sent / word) - 15.8;
    printf("Letter(s) %i\n", let);
    printf("Word(s) %i\n", word);
    printf("Sentence(s) %i\n", sent);
    if (grade <= 1)
    {
        printf("Before Grade 1\n");
    }
    else if (grade < 16)
    {
        printf("Grade %i\n", grade);
    }
    else
    {
        printf("Grade 16+\n");
    }
}

// int index(float l, float s, float w)
// {
//     int i = round(0.0588 * (100 * l / w) - 0.296 * (100 * s / w) - 15.8);
//     return i;
// }

File: readability.py
from cs50 import get_string

text = get_string('Text: ')
i = 0
let = 0
word = 1
sent = 0
for i in range(len(text)):
    if text[i].isalpha():
        let += 1
    elif text[i] == ' ' and text[i + 1]:
        word += 1
    elif text[i] == '?' or text[i] == '!' or text[i] == '.':
        sent += 1

L = float(let / word * 100)
S = float(sent / word * 100)
index = round((float)(0.0588 * L - 0.296 * S - 15.8))
if (index >= 16):
    print('Grade 16+')
elif (index < 1):
    print('Before Grade 1')
else:
    print('Grade', index)